sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -181,18 +191,30 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
     require(versionToBytes(version).sameElements(idToBytes(mod.parentId)),
       s"Incorrect state version!: ${mod.parentId} found, " + s"$version expected")
 
-    if(hasCeased) {
+    if (hasCeased) {
       throw new IllegalStateException(s"Can't apply Block ${mod.id}, because the sidechain has ceased.")
     }
     val consensusEpochNumber = TimeToEpochUtils.timeStampToEpochNumber(params, mod.timestamp)
 
-    val currentWithdrawalEpochInfo = stateStorage.getWithdrawalEpochInfo.getOrElse(WithdrawalEpochInfo(0,0))
+    val currentWithdrawalEpochInfo = stateStorage.getWithdrawalEpochInfo.getOrElse(WithdrawalEpochInfo(0, 0))
     val modWithdrawalEpochInfo = WithdrawalEpochUtils.getWithdrawalEpochInfo(mod.mainchainBlockReferencesData.size, currentWithdrawalEpochInfo, params)
 
     validateBlockTransactionsMutuality(mod)
-    mod.transactions.foreach(tx => validate(tx, consensusEpochNumber, modWithdrawalEpochInfo.epoch).get)
+    mod.transactions.foreach(tx => {
+      validate(tx, consensusEpochNumber, modWithdrawalEpochInfo.epoch).get
+      tx match {
+        case transaction: AbstractCrossChainRedeemTransaction =>
+          crossChainValidatorContainer.validateReceivingCrossChain(
+            new CrossChainRedeemMessageBodyToValidate(transaction)
+          )
+        case _ =>
+      }
+    })
+    crossChainValidatorContainer.validateSenderCrossChain(
```

Comment:
with the above suggestion this piece becomes something like:
sc2scBlockValidators.foreach ( val.validate(mod))